test/test_client.py

```diff
@@ -1591,6 +1591,25 @@ def test_network_error_message(self):
             with self.assertRaisesRegex(AutoReconnect, expected):
                 client.pymongo_test.test.find_one({})
 
+    def test_process_periodic_tasks(self):
```

Comment:
I suspect this test will fail on PyPy because garbage collection works slightly differently. Let's skip this test entirely when running with PyPy (see my current PR for how to do that)

Reply:
Done.